inc/dropdown.class.php

```diff
@@ -301,33 +301,57 @@ static function getDropdownName($table, $id, $withcomment = 0, $translate = true
       $comment = "";
 
       if ($id) {
-         $SELECTNAME    = "'' AS transname";
-         $SELECTCOMMENT = "'' AS transcomment";
-         $JOIN          = '';
+         $SELECTNAME    = new \QueryExpression("'' AS ". $DB->quoteName('transname'));
+         $SELECTCOMMENT = new \QueryExpression("'' AS " . $DB->quoteName('transcomment'));
+         $JOIN          = [];
+         $JOINS         = [];
          if ($translate) {
             if (Session::haveTranslations(getItemTypeForTable($table), 'name')) {
-               $SELECTNAME = "`namet`.`value` AS transname";
-               $JOIN       .= " LEFT JOIN `glpi_dropdowntranslations` AS namet
-                                 ON (`namet`.`itemtype` = '".getItemTypeForTable($table)."'
-                                     AND `namet`.`items_id` = `$table`.`id`
-                                     AND `namet`.`language` = '".$_SESSION['glpilanguage']."'
-                                     AND `namet`.`field` = 'name')";
+               $SELECTNAME = new \QueryExpression("namet.value AS ". $DB->quoteName('transname'));
+               $JOINS['glpi_dropdowntranslations AS namet'] = [
+                  'ON' => [
+                     'namet'  => 'items_id',
+                     $table   => 'id', [
+                        'AND' => [
+                           'namet.itemtype'  => getItemTypeForTable($table),
+                           'namet.language'  => $_SESSION['glpilanguage'],
+                           'namet.field'     => 'name'
+                        ]
+                     ]
+                  ]
+               ];
             }
             if (Session::haveTranslations(getItemTypeForTable($table), 'comment')) {
-               $SELECTCOMMENT = "`namec`.`value` AS transcomment";
-               $JOIN          .= " LEFT JOIN `glpi_dropdowntranslations` AS namec
-                                    ON (`namec`.`itemtype` = '".getItemTypeForTable($table)."'
-                                        AND `namec`.`items_id` = `$table`.`id`
-                                        AND `namec`.`language` = '".$_SESSION['glpilanguage']."'
-                                              AND `namec`.`field` = 'comment')";
-            }
-
-         }
-
-         $query = "SELECT `$table`.*, $SELECTNAME, $SELECTCOMMENT
-                   FROM `$table`
-                   $JOIN
-                   WHERE `$table`.`id` = '$id'";
+               $SELECTCOMMENT = new \QueryExpression("namec.value AS " . $DB->quoteName('transcomment'));
```

Comment:
I think the usage of `QueryExpression` is useless here.
```suggestion
               $SELECTCOMMENT = 'namec.value AS transcomment';
```